### ISSUE TITLE
MSBuild parser does not recognize warnings after JENKINS-56613

### ISSUE THREAD

**Issue by bluejack2000** [jenkinsci/analysis-model#1476]
### Jenkins and plugins versions report

```
Jenkins: 2.528.2
OS: Linux - 6.1.0-30-amd64
Java: 21.0.9 - Eclipse Adoptium (OpenJDK 64-Bit Server VM)
analysis-model-api:13.18.0-935.v784ca_107400a_
...
```

Since November 2025 we have the problem, that the MSBuild parser does not work properly anymore. Since we updated from 3.11 to 3.15 this problem exists and seems to be related to #1330 . I have let ChatGPT analyze the problem and think it detected a problem in the implementation. Here is the bugreport:

### Summary

After upgrading `analysis-model-api` from `13.11.0-919.vda_c75b_c73b_cc` to `13.15.0-928.v63f6dc44cfd2`, MSBuild warnings are no longer detected by Warnings NG in our Jenkins jobs.

`warnings-ng` itself stayed unchanged:
- working:
  - warnings-ng: `12.9848.v5603977006f2`
  - analysis-model-api: `13.11.0-919.vda_c75b_c73b_cc`
- broken:
  - warnings-ng: `12.9848.v5603977006f2`
  - analysis-model-api: `13.15.0-928.v63f6dc44cfd2`

So this looks like a regression in the underlying MSBuild parser shipped via `analysis-model-api`.

### What we see

The Jenkins build log clearly contains valid MSBuild warning lines, but no warnings are reported anymore.

Example line from the persisted raw job log:

```
[2025-11-24T09:30:26.235Z]     34>d:\****\workspace\MPA_master\ramdisk\mpa-clients\target\sources\MPAClient\WPF\Services\WpfMessageBoxService.cs(26,57): warning CS8625: Ein NULL-Literal kann nicht in einen Non-Nullable-Verweistyp konvertiert werden. [d:\****\workspace\MPA_master\ramdisk\mpa-clients\target\sources\MPAClient\MPAClient_egz53oxb_wpftmp.csproj]^M
```

### Expected behavior

That line should be parsed as regular MSBuild warning, just like they were with `analysis-model-api 13.11.0-919.vda_c75b_c73b_cc`.

### Actual behavior

No warning is created at all.

### Suspected root cause

This looks like a regression in `MsBuildParser` between the versions above.

Our raw Jenkins logs contain file paths with `*`, e.g.:

`d:\****\workspace\...`

In the newer parser, paths containing `*` appear to be treated as invalid path names, normalized to `"-"`, and then discarded as tool names / invalid file names. As a result, otherwise valid MSBuild warning lines are dropped completely.

This would explain why:
- the exact same Jenkins job worked before with `analysis-model-api 13.11.0`
- it stopped working immediately after updating only `analysis-model-api`
- `warnings-ng` itself did not change

### Reproduction idea

A minimal reproduction should be possible with an MSBuild warning line like this:

```
[2025-11-24T09:30:26.235Z]     34>d:\****\workspace\foo\Bar.cs(26,57): warning CS8625: Example warning text [d:\****\workspace\foo\foo.csproj]^M
```

If the parser treats the `*` inside the path as invalid and rejects the issue, the warning disappears.

### Notes

The `****` are present in the persisted raw Jenkins job log, not only in browser output.

Could you please check whether the MSBuild parser should still accept such lines, or at least avoid discarding the whole issue when masked/redacted paths contain `*`?

**Comment by bluejack2000** [comment]
Looks like that the folder name `jenkins` which we use is also a credential which is masked out by default. So everytime a credential is found in a path name, it will be masked with **** in the log.

**Comment by uhafner** [comment]
@bluejack2000 When we allow warnings with files of the format `C:\***\file.txt` then those file produce errors later on while resolving the files. Did this not happen in the previous version? Or did you simply ignore those errors? 

Or does the workaround in the [previous comment](https://github.com/jenkinsci/analysis-model/issues/1476#issuecomment-4176069953) help?

**Comment by uhafner** [comment]
@bluejack2000 Do you have any comments? Or can I close this issue as there is a workaround available, see [comment from  Kalle](https://github.com/jenkinsci/analysis-model/issues/1476#issuecomment-4176069953).

**Comment by bluejack2000** [comment]
@uhafner 
Sorry for my late reply.

We use the workaround at the moment. That means the bug has no high prio for us.

You asked how it looked before. We had all warnings registered and the paths worked for us. No problem that they started with ***.

<img width="703" height="500" alt="Image" src="https://github.com/user-attachments/assets/159208b0-8bed-42f6-9c88-b60cd25d84a5" />

<img width="765" height="442" alt="Image" src="https://github.com/user-attachments/assets/4387a1d8-39e6-4574-98d5-da2393c520a4" />

<img width="1033" height="563" alt="Image" src="https://github.com/user-attachments/assets/5cddb8d2-ad3d-4bc8-b3d3-d62d9295ab5f" />

**Comment by uhafner** [comment]
The last screenshot indicates that the path resolution does not work and that the source files could not be read (otherwise the file `AccountAssigmentLabel.cs` would be a link to the source code). Don't you have errors in the warnings info page? 

Example: https://ci.jenkins.io/job/Plugins/job/analysis-model/job/main/853/eslint/info/

You need to use the "info" URL for the warnings.

**Comment by bluejack2000** [comment]
Yes we see errors in the info Page.

```
Can't create fingerprints for some files:
- 'D:/****/workspace/MPA_master/ramdisk/mpa-clients/target/sources/MPAClientCore/Properties/AssemblyVersionInfo.cs', IO exception has been thrown: java.nio.file.InvalidPathException: Illegal char <*> at index 3: D:/****/workspace/MPA_master/ramdisk/mpa-clients/target/sources/MPAClientCore/Properties/AssemblyVersionInfo.cs
```

We never had the need to see the source code or were bothered with these plugin warnings, so the previous behavior of reporting the file with line number was completely ok for us. We were just surprised to see no warning anymore after updating the plugin. That was unexpected and took some time to find out.
